core/src/ast/mod.rs

```diff
@@ -206,20 +207,32 @@ impl ToSql for Statement {
                 source,
                 engine,
                 foreign_keys,
+                primary_key,
                 comment,
             } => {
                 let if_not_exists = if_not_exists.then_some("IF NOT EXISTS");
+
+                let primary_key = primary_key
+                    .as_ref()
+                    .map(|cols| format!("PRIMARY KEY ({})", cols.join(", ")));
                 let body = match (source, columns) {
                     (Some(query), _) => Some(format!("AS {}", query.to_sql())),
                     (None, None) => None,
                     (None, Some(columns)) => {
                         let foreign_keys = foreign_keys.iter().map(ToSql::to_sql);
-                        let body = columns
+                        let mut body = columns
                             .iter()
                             .map(ToSql::to_sql)
                             .chain(foreign_keys)
-                            .collect::<Vec<_>>()
-                            .join(", ");
+                            .collect::<Vec<_>>();
+
+                        // If we identified earlier that there is a primary key, we add it
+                        // at the end of the CREATE TABLE statement.
+                        if let Some(primary_key) = primary_key {
+                            body.push(primary_key);
+                        }
+
+                        let body = body.join(", ");
```

Comment:
nit. Option implement IntoIterator so, we can have more simple version.
```rust
let body = columns
    .iter()
    .map(ToSql::to_sql)
    .chain(foreign_keys)
    .chain(primary_key)
    .collect::<Vec<_>>()
    .join(", ");
```

Reply:
Cleaned up code as described